### PR TITLE
Be upfront about what you do not support

### PR DESCRIPTION
**Pull request by rugk**
You've declined https://github.com/2factorauth/twofactorauth/pull/3547 and in https://github.com/2factorauth/twofactorauth/pull/3546 you've explained:
> We don’t list self hosted sites/services.

However, that was actually **stated nowhere in your guidelines or so**! I mean this is your website/your rules, so it's totally your right to say what to support, but then you should be honest about what you not want to list here.

Without any indication of that you are essentially **wasting the time** of contributors. If they can read in the Contributing.md that self-hosted services are not supported, okay, they won't bother to create a PR here. Otherwise they'll do and this results in things like https://github.com/2factorauth/twofactorauth/pull/3546.

### CODE REVIEW

**Comment by rugk** [comment]
And before you try to argue your "Alexa top 200k" already covers that: It [does not](https://www.alexa.com/siteinfo/mastodon.social) as the largest Mastodon instance (/cc @gargron) e.g. does easily fulfill this criteria. And this is just one – theoretically you have to sum up all different instances and their rank. (respectively not sum up, but consider them all – you get what i mean :wink:)

**Comment by Gargron** [comment]
Not listing self-hosted software seems odd. If it's popular, fulfills the same function as centralized alternatives, and has two-factor authentication, why should it not be listed? If you can't list the software in general, how about listing individual Mastodon servers, since they are websites? E.g. mastodon.social like @rugk points out above.

**Comment by rugk** [comment]
> how about listing individual Mastodon servers, since they are websites? E.g. mastodon.social like @rugk points out above.

However, from a decentralized point of view (or "federation" view) that also seems a little unfair/bad, as obviously _all_ instances/servers that run this software (can) support it.
Also, in other cases like Nextcloud there is no "one big server" (at least not that I know of), but I'd guess all instances together easily would be in the 200k Alexa list.

**Comment by jomo** [comment]
What's the definition of self-hosted? Aren't Bugzilla or GitLab self-hosted?

**Comment by rugk** [comment]
Any news here?

FYI [this fork of this project has merged Nextcloud](https://github.com/Nitrokey/dongleauth/pull/167#issuecomment-4399274799). Just so you know…

**Comment by kinafu** [comment]
For example Nextcloud:

You do not have to host the service by yourself. There are companies/people providing instances to the public. You'll just need to register.

There is _no difference_  for a user between accessing Nextcloud or Dropbox. No additional challenges for the user.

The services vary in their licence and the dev. process (open source), but that's another layer of view.
The user just wants to know about 2fa when visiting this web page.

Self-hosted? "I want cloud and 2fa" :smile:

**Comment by Gargron** [comment]
>You do not have to host the service by yourself. There are companies/people providing instances to the public. You'll just need to register.

The same is true for Mastodon.

**Comment by rugk** [comment]
**TL;DR:** Upvote https://github.com/2factorauth/twofactorauth/issues/3476. :wink:

Any news here? Could not you at least _explain_ why you don't want to include self-hosted services…?

Or answer the questions that have been asked here several times e.g.:
* Why no self-hosted?
* Where do you draw the line, i.e.:
* "What's the definition of self-hosted? Aren't Bugzilla or GitLab self-hosted?"
* Why not include e.g. mastodon.social?
* etc.

On that topic here also saw https://github.com/2factorauth/twofactorauth/issues/3030 (by @drohhyn and @mfc – the latter also having raised concerns about the documentation here).
Also saw https://github.com/2factorauth/twofactorauth/issues/1592 about this topic, where some discussion sparked, but I also did not see any conclusion or so there…
The issue has been raised by @kilasuit in https://github.com/2factorauth/twofactorauth/issues/3234 and I really found this old issue by myself (https://github.com/2factorauth/twofactorauth/issues/1296) about a closed-source self-hosted forum software.

Also there is always the saying "we do not list self-hosted software", and sometimes you even point to the Contributing.md, which makes no sense, as this actually _is not_ documented in that file. You can merge this PR, of course, to change that – or re-evaluate the whole decision.

---

Finally, seeking through all these scattered issues, I found https://github.com/2factorauth/twofactorauth/issues/3476 (I hope I've cross-linked enough to make it easy to find them now :wink:), which looks promising, but has not been received many replies.

**Comment by Carlgo11** [comment]
@rugk I haven't reviewed your PR yet as I'm working on listing all excluded categories in EXCLUSION.md.
Depending on how that turns it it might be better to have this text over there instead.  

Regarding your question why list self-hosted sites aren't listed; It began as a way to keep our lists somewhat small and manageable. There are billions of self-hosted services that pop up and just as quickly disappear.
I suppose this is somewhat fixed by the 200k alexa rule as mentioned in #3476.
Another reason is that TwoFactorAuth.org is targeted towards the consumer and not webmasters. (See our readme)
It's very unlikely that a general consumer would quit a social media platform because said platform don't support 2FA and instead host their own Mastodon server, or host their own ownCloud server because their cloud storage provider don't support 2FA.  


I use self hosted stuff a lot. I have my own mail server and cloud storage using self-hosted services but I'm not a general consumer. And I'm guessing neither are you.

If you grabbed someone randomly off the streets and explained to them about the benefits of 2FA and the risks of not having it I'm very skeptic that you'd manage to get them to host their own things.

Anyways, I might be wrong here. @RichJeanes suggested we'd remove the self-hosting exception in his issue over at #3476. You should vote on it!  

By the way, I'm genuinely curious were all of you came from. Every time a self-hosted thing is mentioned here, there's always a bunch of new people commenting. Like a bot army just waiting to pitch in.
@rugk is the only one here that's not new so I'm very curious how everyone else found this PR.

//Carl

**Comment by jancborchardt** [comment]
@Carlgo11 well, I am designer on Nextcloud and @Gargron works on Mastodon, so we are definitely not bots. :wink: (Sure, we are biased too. ;)

Some of our points are:
- You are essentially running a site for awareness about security and privacy. _Excluding_ things like Mastodon and Nextcloud which seem to fit the bill perfectly just because there are multiple providers seems odd.
- Both Mastodon and Nextcloud, while also possible to self-host, have easy ways to sign up with providers. Mastodon at https://joinmastodon.org/#getting-started and Nextcloud at https://nextcloud.com/signup/
- There are a whole lot of Cryptocurrencies listed, so you could argue that’s also quite nerdy and not easily understandable if you "grabbed someone randomly off the streets". ;) Same for "Cloud computing" and the "Developer" categories.
- You already list open source apps like Wordpress (with their .com instance) and Minds (with their .com instance).

Hope that makes it more understandable why it feels strange that among these rules, Mastodon and Nextcloud are excluded even though they are quite popular, and arguably less nerdy and/or easier to sign up to for a "general consumer" than a lot of other services on this list.

**Comment by Gargron** [comment]
I understand it's frustrating when a lot of people who have never appeared in your repository before come in with opinions. Anyway, we're here because this issue was discussed on Mastodon, a decentralized social network with close to 2 million users spread over around 2,000 servers.

**Review by mxxcon — COMMENTED** [review]
@rugk I take a problem with the title of your pull-request. Your implication that we some how lie about what gets listed is extremely inaccurate and offensive if not to all the contributors, then at least to me!

It has been a long-standing problem that nobody seems to  be willing to help addressing but only complain about.
It's been discussed multiple times in the past. If you can come up with a way for us to accurately yet easily represent 2fa status of self-hosted projects in all the possible different combinations that can arise from the use of external plugins and custom configurations then we'll be more than happy to re-evaluate the current guideline.

**Comment by rugk** [comment]
You may not have lied, and I don't want to claim you did, but you just miss the information this pull requests adds. If you think, this is a way of lying, then this is your interpretation. In any case, you can see this PR just adds (missing) information.
If you don't think so, try to explain how one would see the requirement of "not include self-hosted sites" without the PR.

---

As for the whole discussion about the way you may include 2FA sites, I think https://github.com/2factorauth/twofactorauth/issues/3476 is the way to go. I've written down ideas, there.

**Comment by jancborchardt** [comment]
@rugk just to be a bit more friendly, maybe change the "honest" in the title to "upfront". :slightly_smiling_face:

**Comment by kilasuit** [comment]
@mxxcon - you've been in the past quite against this being opened up to state that self hosted items like Wordpress can *BE* configured to use MFA by **insisting** that it needs to give **Exact** Guidelines as how to do this.

Whereas I would prefer it suggest that these types of sites/applications/etc can be configured to allow use of MFA services and have that show as a potential link away from this site to the providers help pages (if given directly by the provider)

Which I've suggested prior as a solution as per #3234

**Comment by mxxcon** [comment]
> 
> 
> @mxxcon - you've been in the past quite against this being opened up to state that self hosted items like Wordpress can _BE_ configured to use MFA by **insisting** that it needs to give **Exact** Guidelines as how to do this.

I have not insisted that it needs to give exact guidelines _how_ to enable 2fa, but rather that we need to have a way to accurately represent for all the information that would arise from listing such entries.

For example, if we list Mastodon project, do we list the project proper itself or a ton of different specific servers running it? If we list the project proper, but the specific instance doesn't have 2fa enable, then the info on this site is not accurate for that user. If we list specific implementations, many of them will be under 200k alexa rank and won't be listed anyway.

**Comment by rugk** [comment]
> If we list the project proper, but the specific instance doesn't have 2fa enable, then the info on this site is not accurate for that user.

You can have a "note" for that. Or in your example, that's not a concern, because Mastodon enables 2FA by default. (I don't even know if you can disable it – can you, @Gargron?)

IMHO, BTW, yes do list the "project itself".

And, as said thousand of times in this thread already, you already list some of them, i.e. Bugzilla and GitLab. You can also self-host them.

**Comment by kmpoppe** [comment]
Closing this PR in favor of #3632.
